### PR TITLE
homepage functionality

### PR DESCRIPTION
**Pull request by joeyswafford**
added ability to change zipcode.

added categories to sidenav which gets pulled from database. Right now the categories are hard-coded. That can be deleted when the backend feature is added.

added ability to login and logout. Review homeRoutes.js

### CODE REVIEW

File: controllers/homeRoutes.js [hunk 1]
```diff
@@ -139,5 +154,10 @@ router.get('/login', (req, res) => {
   res.render('login');
 });
 
+router.get('/logout', (req, res) => {
+  req.session.logged_in = false
+  res.render('homepage',{ ...dataObject, logged_in: req.session.logged_in});
+});
+
```
Comment (WillBerner) [inline controllers/homeRoutes.js:161]: We actually already have a logout route in /controllers/api/userRoutes (which also makes sure a user is logged in before logging them out, although idk how that could be abused).

File: views/layouts/main.handlebars [hunk 2]
```diff
@@ -10,6 +10,7 @@
   <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.3/css/bulma.min.css" />
   <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400&display=swap" rel="stylesheet" />
   <link rel="stylesheet" href="/css/style.css"/>
+  <script src="/js/utils.js"></script>
```
Comment (WillBerner) [inline views/layouts/main.handlebars:13]: I don't see the file this is referencing anywhere.

File: controllers/homeRoutes.js (hunk 1 above)
Comment (joeyswafford) [reply controllers/homeRoutes.js]: Repurposed this to fetch the backend logout request.

**Review by WillBerner — APPROVED** [review]
LGTM

**Pull request merged**